### ISSUE TITLE
Use different ES instance

### ISSUE THREAD

**Issue by tgjorgoski** [aws-amplify/amplify-cli#4591]
Is there a way to somehow update the generated environment, so to use different ES instance?
I tried switching manually the source in the AppSync, and then changing the ES_INSTANCE env. variable of the streaming es python lambda, and it worked well.
However the problem is that it seems that amplify push doesn't work after that.
Any suggestions?

**Comment by undefobj** [comment]
Yes here you go: https://docs.amplify.aws/cli/graphql-transformer/config-params#elasticsearchinstancetype

**Comment by tgjorgoski** [comment]
Sorry if I wasn't clear, I didn't ask about using different type of instance, but actually using another instance.

**Comment by tgjorgoski** [comment]
Can you please reopen the ticket, so someone might answer?

**Comment by undefobj** [comment]
Could you explain more what you're trying to accomplish? It's still unclear.

**Comment by tgjorgoski** [comment]
Thanks, sure! 

I'm trying to reuse the same elasticsearch instance in multiple environments. (mainly because of the costs which are raising a lot when you have lot of environments and one ES per environment).
 
I checked the open tickets here, and I found several which raise a similar issue, but it seems  there is no easy way to do it. So I thought I try to do it "manually". What I did is the following: I created second environment, and did amplify push. 
After that I manually went into the AppSync aws console for that new environment and modified the data-source for the ES to use the ES instance from an old environment. I also went to the aws console for the streaming es lambda for the new environment and set the ES_ENDPOINT environment variable to point to the old ES instance, and modified permissions to give it access to the old ES instance.

When I did those changes, everything worked as I wanted - I had two environments which were using one and the same ES instance. Even after deleting the new ES instance created by the second environment, everything was working fine.

HOWEVER the problem is that when I try to do "amplify push" for any new changes to this second environment, the operation fails with the "Following resources failed" for all of search related resolvers.

I'm trying to understand if there is any change ( be it modification of CFNs, or something else), which would help me with this setup.

**Comment by attilah** [comment]
@tgjorgoski resource reuse is something that will be addressed in an upcoming RFC that you can find here: https://github.com/aws-amplify/amplify-cli/issues/3977

I mark the issue as an enhancement and will track as part of that RFC and see if this can be included in an earlier stage within that.
